lib/postgrex/protocol.ex

```diff
@@ -654,18 +675,28 @@ defmodule Postgrex.Protocol do
     end
   end
 
+  defp execute_send( s, status, %Stream{ state: :suspended, portal: portal, max_rows: max_rows, query: query } = stream, _params, buffer ) do
+    %Query{ name: name } = query
+    messages = [ msg_execute(name_port: portal, max_rows: max_rows) | sync_msgs( status, name ) ]
+    send_and_recv( s, status, stream, buffer, messages, &execute_recv/4 )
```

Comment:
I extracted `send_and_recv` because I felt some duplication with two clauses 

```
defp execute_send( s, status, %Stream{ state: :suspended
```

and 

```
defp execute_send( s, status, %Stream{ portal: portal, 
```

the diff was the function called, not it's arity, hence the _easy_ extract

I'll push in a few minutes a new pile to try this kind of change ( if I understand )